### PR TITLE
Flush Go Cache Each Month

### PR DESCRIPTION
**Pull request by davidvonthenen**
## What this PR does / why we need it

Currently it seems like we keep the Go Cache around forever. This leads to very long restore times when trying to load the Go Cache. A lot of this stuff might be very stale and we don't want to carry that stuff around. This is a compromise so that we flush out and store the go cache each month. Picking that timeframe because we would like to do monthly releases and it would seem to cycle out old dependencies each month.

## Which issue(s) this PR fixes

NA

## Describe testing done for PR

NA

## Special notes for your reviewer

NA

### CODE REVIEW

File: .github/workflows/check-lint.yaml [hunk 1]
```diff
@@ -26,11 +26,19 @@ jobs:
...
 
-      - name: Cache linting
-        uses: actions/cache@v2
+      - name: Get Date
+        id: get-date
+        shell: bash
+        run: |
+          echo "::set-output name=date::$(date -u "+%Y-%m")"
+
+      - name: Restore Go Cache
```
Comment (stmcginnis) [inline .github/workflows/check-lint.yaml:35]: nit: This one isn't a go cache.

**Review by stmcginnis — DISMISSED** [review]
Looks good. If I understand right, the first time each action runs at the start of a new month, it will try to restore the cache but not find it. So a new cache will be created and subsequent runs for the rest of the month will be able to take advantage of the cached data. Then things get reset again the next month, so we don't end up accumulating more and more data in the cache.

**Comment by dvonthenen** [comment]
> Looks good. If I understand right, the first time each action runs at the start of a new month, it will try to restore the cache but not find it. So a new cache will be created and subsequent runs for the rest of the month will be able to take advantage of the cached data. Then things get reset again the next month, so we don't end up accumulating more and more data in the cache.

That's absolutely right. Because the cache key is tied to "YYYY-MM" in the key.

File: .github/workflows/check-lint.yaml (hunk 1 above)
Comment (dvonthenen) [reply .github/workflows/check-lint.yaml:35]: Ooops. Copy and paste. Will fix

**Review by stmcginnis — APPROVED** [review]

**Pull request merged**